Cruxite blocks

glubtier opened the pull request:
Added cruxite block set, derivative blocks, and recipes.

Textures:
raw block - Rose
smooth block - Sipher
Polished, bricks, chiseled, door, trapdoor, lamp, pillar - Glubtier

kirderf1 commented on src/main/java/com/mraof/minestuck/block/CustomLampBlock.java:
```diff
@@ -0,0 +1,38 @@
+	public InteractionResult use(BlockState pState, Level pLevel, BlockPos pPos,
+								 Player pPlayer, InteractionHand pHand, BlockHitResult pHit) {
+		if(!pLevel.isClientSide() && pHand == InteractionHand.MAIN_HAND) {
+			boolean currentState = pState.getValue(CLICKED);
+			pLevel.setBlock(pPos, pState.setValue(CLICKED, !currentState), 3);
```
```suggestion
			pLevel.setBlock(pPos, pState.cycle(CLICKED), 3);
```

kirderf1 replied:
I suggest using the named constant `Block.UPDATE_ALL` in place of that number literal.

kirderf1 commented on src/main/java/com/mraof/minestuck/data/recipe/MinestuckRecipeProvider.java:
```diff
@@ -269,6 +269,50 @@ protected void buildRecipes(Consumer<FinishedRecipe> recipeBuilder)
 		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.PERFECTLY_GENERIC_TRAPDOOR.get(), 2).define('#', MSBlocks.GENERIC_OBJECT.get()).pattern("###").pattern("###").group("wooden_trapdoor").unlockedBy("has_generic_object", has(MSBlocks.GENERIC_OBJECT.get())).save(recipeBuilder);
 		
 		
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_STAIRS.get(), 4).define('#', MSBlocks.CRUXITE_BLOCK.get()).pattern("#  ").pattern("## ").pattern("###").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_SLAB.get(), 6).define('#', MSBlocks.CRUXITE_BLOCK.get()).pattern("###").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_WALL.get(), 6).define('#', MSBlocks.CRUXITE_BLOCK.get()).pattern("###").pattern("###").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapelessRecipeBuilder.shapeless(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_BUTTON.get()).requires(MSBlocks.CRUXITE_BLOCK.get()).unlockedBy("has_raw_cruxite", has(MSItems.RAW_CRUXITE.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_PRESSURE_PLATE.get(), 1).define('#', MSBlocks.CRUXITE_BLOCK.get()).pattern("##").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_DOOR.get(), 3).define('#', MSBlocks.SMOOTH_CRUXITE_BLOCK.get()).pattern("## ").pattern("## ").pattern("## ").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_TRAPDOOR.get(), 2).define('#', MSBlocks.SMOOTH_CRUXITE_BLOCK.get()).pattern("###").pattern("###").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.POLISHED_CRUXITE_BLOCK.get(), 4).define('#', MSBlocks.SMOOTH_CRUXITE_BLOCK.get()).pattern("## ").pattern("## ").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.POLISHED_CRUXITE_STAIRS.get(), 4).define('#', MSBlocks.POLISHED_CRUXITE_BLOCK.get()).pattern("#  ").pattern("## ").pattern("###").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.POLISHED_CRUXITE_SLAB.get(), 6).define('#', MSBlocks.POLISHED_CRUXITE_BLOCK.get()).pattern("###").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.POLISHED_CRUXITE_WALL.get(), 6).define('#', MSBlocks.POLISHED_CRUXITE_BLOCK.get()).pattern("###").pattern("###").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_BRICKS.get(), 4).define('#', MSBlocks.CRUXITE_BLOCK.get()).pattern("## ").pattern("## ").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_BRICK_STAIRS.get(), 4).define('#', MSBlocks.CRUXITE_BRICKS.get()).pattern("#  ").pattern("## ").pattern("###").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_BRICK_SLAB.get(), 6).define('#', MSBlocks.CRUXITE_BRICKS.get()).pattern("###").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_BRICK_WALL.get(), 6).define('#', MSBlocks.CRUXITE_BRICKS.get()).pattern("###").pattern("###").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		SimpleCookingRecipeBuilder.smelting(Ingredient.of(MSBlocks.CRUXITE_BLOCK.get()), RecipeCategory.BUILDING_BLOCKS, MSBlocks.SMOOTH_CRUXITE_BLOCK.get(), 0.1f, 200).group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSItems.RAW_CRUXITE.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CHISELED_CRUXITE_BLOCK.get(), 4).define('#', MSBlocks.CRUXITE_BRICKS.get()).pattern("## ").pattern("## ").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_PILLAR.get(), 2).define('#', MSBlocks.SMOOTH_CRUXITE_BLOCK.get()).pattern("#  ").pattern("#  ").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_LAMP.get(), 1).define('#', MSItems.RAW_CRUXITE.get()).define('%', Items.TORCH).pattern("###").pattern("#%#").pattern("###").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSBlocks.WHITE_CHESS_BRICKS.get())).save(recipeBuilder);
```
These are using `WHITE_CHESS_BRICKS` for the recipe unlock condition.

glubtier replied:
Not sure how that slipped through, when I used a find/replace... copy paste error? idk I'll update these.

kirderf1 reviewed (changes requested):
By raw block, I assume you mean the original `cruxite_block` one.
You should probably add the retexture of that block to the changelog. You should also plan with akisephila how the cruxite block retexture conflict between this and #572 should be handled.

glubtier commented:
> By raw block, I assume you mean the original `cruxite_block` one. You should probably add the retexture of that block to the changelog. You should also plan with akisephila how the cruxite block retexture conflict between this and #572 should be handled.

Correct, it's just called "CRUXITE_BLOCK" in code but they're all cruxite blocks, which is why I specified.

I think we both have the same file there, if Aki pulled from Rose's branch, but I'll ask.

glubtier commented:
Updated recipes and corrected some conflicts with block tags, but it's failing the build on GitHub now, not sure why? I tried runData again and there's nothing changed, nothing unversioned, etc.

Dweblenod commented:
> Updated recipes and corrected some conflicts with block tags, but it's failing the build on GitHub now, not sure why? I tried runData again and there's nothing changed, nothing unversioned, etc.

In the cases where it fails and you cant see where, open Details!
Heres a part of the error that seems relevant
`/home/runner/work/minestuck/minestuck/src/main/java/com/mraof/minestuck/data/MSBlockStateProvider.java:1766: error: cannot find symbol
			if(state.getValue(CustomLampBlock.CLICKED) ) {
			                  ^
  symbol:   variable CustomLampBlock
  location: class MSBlockStateProvider`

glubtier commented:
> In the cases where it fails and you cant see where, open Details! Heres a part of the error that seems relevant `/home/runner/work/minestuck/minestuck/src/main/java/com/mraof/minestuck/data/MSBlockStateProvider.java:1766: error: cannot find symbol if(state.getValue(CustomLampBlock.CLICKED) ) { ^ symbol: variable CustomLampBlock location: class MSBlockStateProvider`

Alright well I'm bamboozled on this because I didn't change those files at all, and I can't replicate it on my end so I'll have to fiddle with it more tomorrow.

Unrelated to that, I have just now committed a change to MSBlockStateProvider. I forgot I hardcoded CRUXITE_LAMP into customLampWithItem. It is no longer hard coded.

kirderf1 commented:
Ooh, this is a peculiar issue for sure. There are non-conflicting changes that causes the build to fail. Just merge `upcoming-content` and you should see the issue you need to resolve.

glubtier commented:
> Ooh, this is a peculiar issue for sure. There are non-conflicting changes that causes the build to fail. Just merge `upcoming-content` and you should see the issue you need to resolve.

Okay that seemed to work. :)

Dweblenod commented on src/main/java/com/mraof/minestuck/block/CustomLampBlock.java:
```diff
@@ -0,0 +1,37 @@
+
+public class CustomLampBlock extends Block
+{
+	public static final BooleanProperty CLICKED = BooleanProperty.create("clicked");
```
To maintain consistency with our other custom blockstate properties, move the creation of CLICKED to MSProperties!

glubtier replied:
Done

Dweblenod commented on CHANGELOG.md:
```diff
@@ -16,6 +16,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 - New NBT tag on transportalizers that allows them to be locked, permanently preventing the viewing or editing of its id or destination
 - Transportalizer NBT data will now be saved when mined, even without silk touch, and can be viewed on the item's tooltip
 - Added Carved Bush.
+- Cruxite block set
```
Make sure to include sipher in the contributors section!

glubtier replied:
Whoops I put it in the PR but forgot the changelog. 😔 done.

Dweblenod commented on src/main/java/com/mraof/minestuck/data/recipe/MinestuckRecipeProvider.java:
```diff
@@ -271,6 +271,50 @@ protected void buildRecipes(Consumer<FinishedRecipe> recipeBuilder)
+		SimpleCookingRecipeBuilder.smelting(Ingredient.of(MSBlocks.CRUXITE_BLOCK.get()), RecipeCategory.BUILDING_BLOCKS, MSBlocks.SMOOTH_CRUXITE_BLOCK.get(), 0.1f, 200).group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSItems.RAW_CRUXITE.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CHISELED_CRUXITE_BLOCK.get(), 4).define('#', MSBlocks.CRUXITE_BRICKS.get()).pattern("## ").pattern("## ").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSItems.RAW_CRUXITE.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_PILLAR.get(), 2).define('#', MSBlocks.SMOOTH_CRUXITE_BLOCK.get()).pattern("#  ").pattern("#  ").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSItems.RAW_CRUXITE.get())).save(recipeBuilder);
+		ShapedRecipeBuilder.shaped(RecipeCategory.BUILDING_BLOCKS, MSBlocks.CRUXITE_LAMP.get(), 1).define('#', MSItems.RAW_CRUXITE.get()).define('%', Items.TORCH).pattern("###").pattern("#%#").pattern("###").group("cruxite_blocks").unlockedBy("has_raw_cruxite", has(MSItems.RAW_CRUXITE.get())).save(recipeBuilder);
```
Given the capacity to turn it off and on it feels from my perspective like it should use an alternative light source. Perhaps glowstone dust or uranium? I also think it would make sense to use the other forms of cruxite given the sculpted look of the texture! Feel free to make an argument if you want it to stay that way, I see no major game design issues with keeping it this way

glubtier replied:
Hmm I do see what you're saying. My goal was to align it with the lantern, which needs 8 iron nuggets ( -> 1 iron ingot -> 1 iron ore). To account for the ability to turn it on and off, 8 cruxite ore seemed both a logical step up from the lantern and a fair price, considering that the on/off is purely decorative and has to be manually toggled so you can't use it like a redstone lamp. 

I wouldn't be opposed to changing it to 4 uranium + 4 cruxite + torch though. Or maybe 8 cruxite + 1 lantern?
